src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -2097,6 +2098,18 @@ public void listen(EntryAddedEvent addedEntryEvent) {
         }
     }
 
+    private class EntryRemovedListener {
```

Comment:
Why do you introduce this private class? Why not directly add the listener to the BasePanel?

Reply:
Because there are several other listeners in `BasePanel` that are all implemented as separate classes. That way it is easier to one day extract the listeners into different files and reduce the size of `BasePanel`

Btw.: `BasePanel` currently has no closing method, so these listeners are never unregistered and are essentially a memory leak. Not too big a problem, though: A new `BasePanel` is created very rarely and the listeners are very small (another reason for the separate class at the moment).